src/query/api/v1/handler/graphite/find.go

```diff
@@ -57,52 +57,75 @@ func NewFindHandler(
 	}
 }
 
+func mergeTags(
+	noChildren *storage.CompleteTagsResult,
+	withChildren *storage.CompleteTagsResult,
+) (map[string]bool, error) {
+	// sanity check the case.
+	if noChildren.CompleteNameOnly {
+		return nil, errors.New("tags result with no children is completing name only")
+	}
+
+	if withChildren.CompleteNameOnly {
+		return nil, errors.New("tag result with children is completing name only")
+	}
+
+	mapLength := len(noChildren.CompletedTags) + len(withChildren.CompletedTags)
+	tagMap := make(map[string]bool, mapLength)
+
+	for _, tag := range noChildren.CompletedTags {
+		for _, value := range tag.Values {
+			tagMap[string(value)] = false
+		}
+	}
+
+	// NB: fine to overwrite any tags which were present in the `noChildren` map
+	for _, tag := range withChildren.CompletedTags {
+		for _, value := range tag.Values {
+			tagMap[string(value)] = true
```

Comment:
can you just create a set here instead of the true/false thing? I think you can just do something like:
```
tagMap := make(map[string]struct{}, mapLength)
```
then just set it to an empty struct when you're looping through. 
```
		for _, value := range tag.Values {
			tagMap[string(value)] = struct{}{}
		}
```

Reply:
No, I need to see if the given tag has children here so need the bool